source-code/core2/src/lint/api.ts

```diff
@@ -0,0 +1,46 @@
+import type { InlangApp } from "../app/api.js"
+import type { LanguageTag } from "../languageTag.js"
+import type { Message } from "../messages/schema.js"
+
+export type LintRule = {
+	id: `${string}.${string}`
+	displayName: Record<string, string>
+	/**
+	 * The default level of the lint rule.
+	 *
+	 * Can be overwritten by the user config.
+	 */
+	defaultLevel: "error" | "warn"
+}
```

Comment:
## How to make linting easier 

### Problem 

Linting is difficult because the current API tries to be agnostic but idk that leads to difficult implementations

Reply:
The question is will the community use the agnostic linting features more often? Because today, nobody use something other than standard-lint-rules?

Level configuration will become more important with the rise of github actions/ CI. :D

Reply:
+1 lints in general & levels could be more sophisticated.
it's currently loved by our users although it's dead simple.
maybe huge potential in here.  

- not only `string`
- could be object of type `Level` where multiple properties can be exposed

Reply:
> defaultLevel
> could be object of type Level where multiple properties can be exposed

Why? What is the use case?

Reply:
I see a lot of systems going with `severity` like from `1` to `5` in addition to the `level`.
https://about.gitlab.com/handbook/engineering/quality/issue-triage/#severity

but maybe that's not in `defaultLevel` but belongs in `LintRule`.

Reply:
> but maybe that's not in defaultLevel but belongs in LintRule. 

We can always add APIs... please focus on bare minimum surface API that eases the path for future additions and refrain from proposing "nice to have" stuff.

Reply:
Lints should be easy to implement and more used by our users. It's always a big selling point in the first place and the best call to action to improve globalization and use inlang. This should be as easy to create a rule in Grammarly.

Lint rules are used to:
- automate
- communicate between the developer and translator
- define standards for translations

This makes it easier to create and the `displayName` is a good addition 👍 

Quick note: I have the feeling the way we implement linting will be similar to the implementation of collaborative features like flagging translations for a review, as accepted or as blocked by a role, but maybe this shouldn't be mixed up atm.

Reply:
@NiklasBuchfink your reply reads like ChatGPT lol. 

The question is not if we have lint rules, the question is how should the API for lint rules look like.

Reply:
my conclusion: we can always add more properties to `LintRule` to make it more flexible. currently, no need.